src/Symfony/Component/Messenger/Transport/Receiver/SingleMessageReceiver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * Receiver that decorates another, but receives only 1 specific message.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @internal
+ * @experimental in 4.3
+ */
+class SingleMessageReceiver implements ReceiverInterface
+{
+    private $receiver;
+    private $envelope;
+    private $hasReceived = false;
+
+    public function __construct(ReceiverInterface $receiver, Envelope $envelope)
+    {
+        $this->receiver = $receiver;
+        $this->envelope = $envelope;
+    }
+
+    public function get(): iterable
+    {
+        if ($this->hasReceived) {
+            return [];
```

Comment:
since this method uses a generator, it should not return array. but just `return;`

Reply:
actually it does not seem worth to use a generator here, but just return the envolope in an array.

Reply:
#31389